src/proposal.mligo

```diff
@@ -337,18 +337,26 @@ let delete_proposal
     Set.remove (start_date, proposal_key) store.proposal_key_list_sort_by_date
   }
 
+// if proposal is proposed at 0 period_num, it will be expired at 3 period_num
+[@inline]
+let is_proposal_expired (proposal, voting_period, store : proposal * voting_period * storage): bool =
+  let current_period = get_current_period_num(store.start_time, voting_period) in
+  current_period > proposal.period_num + 2n
```

Comment:
Ah my explanation wasn't very clear, when I said:
> a proposal flushed after a certain period (keeping with the previous example `x + 3` for proposals created in `x`, for the same reason as explained above) is dropped.

I meant any period _after_ `x + 3` included, so `x + 4` or later.

My reasoning is that if a proposal is created at `x` then in
- `x + 1` it will be voted on
- `x + 2` it can be flushed, new proposals will be created in this stage but these cannot be flushed yet
- `x + 3` it can be flushed still, the proposals from `x + 2` are being voted on, so they still can't be flushed yet
- `x + 4` is too late: proposals from a newer period `x + 2` can not be flushed too, so the ones from `x` are expired

Reply:
Yep, that's correct.